### PR TITLE
Add extension functions for Ktor plugins 

### PR DESCRIPTION
**Pull request by marychatte**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/10617
Earlier, an [issue](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/10617) about adding extension functions for plugins that allow writing existing code in the Ktor DSL style was opened. 
In this PR, such new functions were added.
These new functions we used in [the example](https://github.com/ktorio/ktor-samples/tree/main/opentelemetry) where also can be found examples of usages new functions for [server](https://github.com/ktorio/ktor-samples/blob/main/opentelemetry/server/src/main/kotlin/opentelemetry/ktor/example/plugins/opentelemetry/setupServerTelemetry.kt) and [client](https://github.com/ktorio/ktor-samples/blob/main/opentelemetry/client/src/main/kotlin/opentelemetry/ktor/example/plugins/opentelemetry/setupClientTelemetry.kt)

### CODE REVIEW

File: instrumentation/ktor/ktor-2.0/library/src/main/kotlin/io/opentelemetry/instrumentation/ktor/v2_0/client/KtorClientTracingBuilder.kt [hunk 1]
```diff
@@ -37,23 +40,104 @@ class KtorClientTracingBuilder {
...
+
+  fun capturedRequestHeaders(headers: Iterable<String>) {
+    setCapturedRequestHeaders(headers.toList())
+  }
+
   fun setCapturedResponseHeaders(vararg headers: String) = setCapturedResponseHeaders(headers.asList())
 
   fun setCapturedResponseHeaders(headers: List<String>) {
     httpAttributesExtractorBuilder.setCapturedResponseHeaders(headers)
   }
```
Comment (trask) [inline instrumentation/ktor/ktor-2.0/library/src/main/kotlin/io/opentelemetry/instrumentation/ktor/v2_0/client/KtorClientTracingBuilder.kt:55]: @marychatte would it make sense to deprecate the `set*()` methods?

Comment (marychatte) [reply instrumentation/ktor/ktor-2.0/library/src/main/kotlin/io/opentelemetry/instrumentation/ktor/v2_0/client/KtorClientTracingBuilder.kt:55]: Yes, thanks. I will do it

**Review by laurit — APPROVED** [review]

**Comment by marychatte** [comment]
Hi! I just wanted to ask if there are any estimates when the PR can be merged?

**Comment by trask** [comment]
hey @marychatte! sorry for the delay, just merged and it will be included in the upcoming release

**Pull request merged**